Source/SVGKImage.h

```diff
@@ -137,6 +144,7 @@ typedef void (^SVGKImageAsynchronousLoadingDelegate)(SVGKImage* loadedImage, SVG
  */
 +(SVGKParser *) imageWithSource:(SVGKSource *)source onCompletion:(SVGKImageAsynchronousLoadingDelegate)blockCompleted;
 
+- (id)initWithContentsOfURL:(NSURL *)url;
```

Comment:
Emm....Because [MaddTheSane's fork](https://github.com/MaddTheSane/SVGKit/blob/master/Source/SVGKImage.h#L133) use this method. And his [SVGImageRep implementation](https://github.com/MaddTheSane/SVGKit/blob/master/Source/AppKit%20additions/SVGKImageRep.m#L130) need this API.

I think maybe this `URL` is just fileURL, but not network URL. Maybe I can remove the public API, use file path instead.

Reply:
Yep, be careful to check that one. SVG is actually specified differnetly by W3C for whether it's loaded from a web URL, or if the URL is being used as a file:// (I think! It's been a long time since I read this).

Reply:
Yeah, it was supposed to be for file URLs, but it could be used for network URLs as well. A disclaimer saying that using it for network URLs might be wise.

Reply:
Keep this. Because currentlly we already have one API `+[SVGKImage imageWithContentsOfURL:]`, even we don't export the `-[SVGKImage initWithContentsOfURL:]` this API. These two should be either available or not at the same time. So I just export the `-[SVGKImage initWithContentsOfURL:]` one. Keep @MaddTheSane code.

Reply:
Looking at the source, initWithContentsOfURL is implemented safely at the moment (it constructs an SVGKSource and fills it out with the URL before proceeding).

Unless I've misread that, I see no reason to avoid having both methods.

(I think there was a time when the + method worked better, because it did some housekeeping, but the - method did not)

Reply:
@adamgit What means `initWithContentsOfURL is implemented safely at the moment`.

The differens between `imageWithContentsOfURL:` and `initWithContentsOfURL:` is because of Objective-C's naming conventions before ARC was introduced. Methods with `init*` should create a object with `retain` to +1 retain count. Methods with `xxxWith*`  should create a autoreleasing object with +0 retain count. See all API provided by apple like [-NSArray array](https://developer.apple.com/documentation/foundation/nsarray/1460120-array?language=objc) && [-NSArray init](https://developer.apple.com/documentation/foundation/nsarray/1414315-init?language=objc).

However, when ARC was introduced, these assume actually take no effect. Because ARC will insert code about retain && release. The autoreleasing object should now be explit set using `@autoreleasing` pool. However, these naming was kept because of large code base. And a class method like `imageWithContentsOfURL:` is just a simply call with `[[xxx alloc] initWithContentsOfURL:]`.

The `@synchronized` here is actually take no effect. I read all function call of this, there are no thread-safe issue to make us using a lock on the object level. The only potential issue is that `globalSVGKImageCache`, a mutable dictionary without any synchonize method. If we want to keep it thread-safe, use `lock` just before any reading-writing operation to that dictionary, or use the thread-safe class [NSCache](https://developer.apple.com/documentation/foundation/nscache). It's no need to put un-related method into `@synchronized`.